app/controllers/CreditsSummaryController.scala

```diff
@@ -90,47 +90,34 @@ class CreditsSummaryController @Inject()(creditsView: CreditsSummary,
                     origin: Option[String] = None)
                    (implicit user: MtdItUser[AnyContent],
                     hc: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
-    if (isDisabled(MFACreditsAndDebits) && isDisabled(CutOverCredits)) {
-      auditCreditSummary(None, Seq.empty)
-      Future.successful(Ok(creditsView(
-        calendarYear = calendarYear,
-        backUrl = if (isAgent) getAgentBackURL(user.headers.get(REFERER), calendarYear) else getBackURL(user.headers.get(REFERER), origin, calendarYear),
-        isAgent = isAgent,
-        utr = user.saUtr,
-        btaNavPartial = user.btaNavPartial,
-        charges = List.empty,
-        maybeAvailableCredit = None,
-        origin = origin))
-      )
-    } else {
-      creditHistoryService.getCreditsHistory(calendarYear, user.nino).flatMap {
-        case Right(credits) =>
-          val charges: List[CreditDetailModel] = ((isEnabled(MFACreditsAndDebits), isEnabled(CutOverCredits)) match {
-            case (true, false) => credits.filter(_.creditType == MfaCreditType)
-            case (false, true) => credits.filter(_.creditType == CutOverCreditType)
-            case _ => credits
-          }).sortBy(_.date.toEpochDay)
-          val maybeAvailableCredit: Option[BigDecimal] =
-            credits.flatMap(_.balanceDetails.flatMap(_.availableCredit.filter(_ > 0.00))).headOption
-          auditCreditSummary(maybeAvailableCredit, charges)
-          Future.successful(Ok(creditsView(
-            calendarYear = calendarYear,
-            backUrl = if (isAgent) getAgentBackURL(user.headers.get(REFERER), calendarYear) else getBackURL(user.headers.get(REFERER), origin, calendarYear),
-            isAgent = isAgent,
-            utr = user.saUtr,
-            btaNavPartial = user.btaNavPartial,
-            charges = charges,
-            maybeAvailableCredit = maybeAvailableCredit,
-            origin = origin)))
-        case Left(_) => {
-          if (isAgent) {
-            Logger("application").error(s"[CreditsSummaryController][showAgentCreditsSummary] - Could not retrieve financial details for Calendar year: $calendarYear, NINO: ${user.nino}")
-            Future.successful(agentItvcErrorHandler.showInternalServerError())
-          }
-          else {
-            Logger("application").error(s"[CreditsSummaryController][showCreditsSummary] - Could not retrieve financial details for Calendar year: $calendarYear, NINO: ${user.nino}")
-            Future.successful(itvcErrorHandler.showInternalServerError())
-          }
+    creditHistoryService.getCreditsHistory(calendarYear, user.nino).flatMap {
```

Comment:
you might need to refactor part related to filtering charges into a separate method under creditHistoryService: something line getCreditsHistoryAndFilter (.. , exclusionPredicate:Seq[CreditType])